design/wallet_extension.md

```diff
@@ -12,35 +12,45 @@ Ethereum wallets (e.g. MetaMask, hardware wallets), tooling (e.g. Remix) and web
     ](https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/ethereum/eth1.0-apis/assembled-spec/openrpc.json)
   * Additional APIs as needed to support common tools (e.g. MetaMask's use of Geth's `net_version` API)
 * The wallet extension is run locally by the end user
-* The wallet extension does not send or receive any sensitive information in plaintext. The following methods are 
-  considered sensitive (this list may grow over time):
-  * `eth_call`
-  * `eth_getBalance`
-  * `eth_getTransactionByHash`
-  * `eth_getTransactionReceipt`
-  * `eth_sendRawTransaction`
 * Encryption
-  * Any encryption keys used are only known to the client (e.g. wallet, webapp), and not to any third-parties
-  * The encryption is transparent to the client; from the client's perspective, they are interacting with a "standard" 
-    non-encrypting implementation of the Ethereum JSON-RPC specification
-* The wallet extension is usable by any webapp, tool or wallet type, and in particular:
-  * Hardware wallets that do not offer a decryption capability
-  * MetaMask, for which the keys are only available when running in the browser
+  * The wallet extension does not send or receive any sensitive information in plaintext. The following methods are 
+    considered sensitive (this list may grow over time):
+    * `eth_call`
+    * `eth_getBalance`
+    * `eth_getTransactionByHash`
+    * `eth_getTransactionReceipt`
+    * `eth_sendRawTransaction`
+  * The keys used to encrypt sensitive information (the *viewing keys*) are stored solely on the client-side (e.g. 
+    wallet, webapp), and are not shared with any third-parties (e.g. the node operator)
+* UX
+  * Any encryption is transparent to the client; from the client's perspective, they are interacting with a "standard" 
+      non-encrypting implementation of the Ethereum JSON-RPC specification
+  * The wallet extension is usable by any webapp, tool or wallet type, and in particular:
+    * Hardware wallets that do not offer a decryption capability
+    * MetaMask, for which the keys are only available when running in the browser
+  * Restarting the wallet extension does not require existing viewing keys to be regenerated
+  * Existing viewing keys are automatically resubmitted if the network is restarted
+  * Multiple wallet extensions can run in parallel, with separate sets of viewing keys and optionally connected to 
+    separate hosts 
 * Some tooling (e.g. MetaMask) does not set the `from` field in `eth_call` requests. For any received `eth_call` 
   request, it is acceptable for the wallet extension to set the `from` field programmatically (e.g. to enable 
   encryption of the response)
 
+## Non-requirements
+
+* Encryption or obfuscation of viewing keys at rest. It is acceptable to store viewing keys locally in the clear
```

Comment:
We need to think about an expiration date for v_k, and about revocation.
Revocation can't just be an rpc call on a node, because the hacker who has the signed viewing keys can always go to another node.
It has to be a transaction of some sort included in the chain.

Adding a date should be relatively easy, as it can be included in the signed payload

Reply:
Added as a requirement.